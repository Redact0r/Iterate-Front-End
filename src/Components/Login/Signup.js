import React, { Component } from "react";
import AuthApiService from "../../services/auth-service";

class Signup extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  handleRegistrationSuccess = (user) => {
    const { history } = this.props;
    history.push("/login");
  };

  state = { error: null };

  handleSubmit = (ev) => {
    ev.preventDefault();
    const {
      signup_fullname,
      signup_nickname,
      signup_username,
      signup_password,
    } = ev.target;
    this.setState({ error: null });
    AuthApiService.postUser({
      user_name: signup_username.value,
      password: signup_password.value,
      full_name: signup_fullname.value,
      nickname: signup_nickname.value,
    })
      .then((user) => {
        signup_fullname.value = "";
        signup_nickname.value = "";
        signup_username.value = "";
        signup_password.value = "";
        this.handleRegistrationSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    return (
      <div>
        <form onSubmit={(e) => this.handleSubmit(e)} className="signup-form">
          <label htmlFor="signup_username">Username</label>
          <input
            id="signup_username"
            type="text"
            className="user signup username"
            placeholder="Username"
          />
          <label htmlFor="signup_fullname">Full Name</label>
          <input
            id="signup_fullname"
            type="text"
            className="user login name"
            placeholder="Full Name"
          />
          <label htmlFor="signup_nickname">Nickname</label>
          <input
            id="signup_nickname"
            type="text"
            className="user signup nickname"
            placeholder="Nickname"
          />
          <label htmlFor="password">Password</label>
          <input
            id="signup_password"
            type="password"
            className="user signup pass"
            placeholder="Password"
          />
          <button className="signup-btn" type="submit">
            Sign Up
          </button>
        </form>
      </div>
    );
  }
}

export default Signup;
