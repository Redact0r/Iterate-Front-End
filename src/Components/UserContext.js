import React, { Component, useState, createContext } from "react";

const UserContext = createContext({
  isLoggedIn: true,
  wordCount: 0,
  DailyWritingGoal: 500,
  streak: 0,
  boxTitle: "",
  boxContent: "",
  daysChecked: {
    M: false,
    T: false,
    W: false,
    R: false,
    F: false,
    S: false,
    U: false,
  },
  goalSelector: {
    Hemingway: true,
    Ballard: false,
    King: false,
    Fox: false,
  },
});

export default UserContext;

// export class UserContextProvider extends Component {
//   state = {
//     isLoggedIn: true,
//     wordCount: 0,
//     DailyWritingGoal: 500,
//     streak: 0,
//     boxTitle: "",
//     boxContent: "",
//     daysChecked: {
//       M: false,
//       T: false,
//       W: false,
//       R: false,
//       F: false,
//       S: false,
//       U: false,
//     },
//     goalSelector: {
//       Hemingway: true,
//       Ballard: false,
//       King: false,
//       Fox: false,
//     },
//   };

//   render() {
//     return (
//       <UserContext.Provider value={this.state}>
//         {this.props.children}
//       </UserContext.Provider>
//     );
//   }
// }
