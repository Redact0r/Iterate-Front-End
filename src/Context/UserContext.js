import React, { Component, createContext } from "react";
import IterateApi from "../services/IterateApi";
import TokenService from "../services/token-service";
import streakService from "../services/streak-service";

const UserContext = createContext({
  isLoggedIn: TokenService.hasAuthToken(),
  userid: null || localStorage.getItem("userid"),
  works: [],
  wordCount: 0,
  animationClassName: "",
  DailyWritingGoal: 500,
  streak: 0,
  currentWorkId: null,
  boxTitle: "",
  boxContent: "",
  daysChecked: {
    M: false,
    T: false,
    W: false,
    R: false,
    F: false,
    S: false,
    U: false,
  },
  goalSelector: {
    Hemingway: true,
    Ballard: false,
    King: false,
    Fox: false,
  },
  handleSave: () => {},
  handleTitleChange: () => {},
  handleContentChange: () => {},
  handleDailyGoalSelector: () => {},
  handleKeypress: () => {},
  handleDayCheck: () => {},
  handleStartNew: () => {},
  handleDelete: () => {},
  handleEdit: () => {},
  handleAnimationKeypress: () => {},
  stopAnimate: () => {},
  generateWorks: () => {},
  handleLogging: () => {},
  setUserId: () => {},
  setStreak: () => {},
  setMessage: () => {},
  resetMessage: () => {},
  setStreakComplete: () => {},
});

export default UserContext;

export class UserContextProvider extends Component {
  state = {
    userid: null || localStorage.getItem("userid"),
    works: [],
    wordCount: 0,
    DailyWritingGoal: 500,
    streak: 0,
    currentWorkId: null,
    boxTitle: "",
    boxContent: "",
    goalSelector: "",
    animationClassName: "",
    message: "",
    isShow: false,
    streakComplete: false,

    handleDayCheck() {
      this.handleDayCheck.bind(this);
    },
    handleEdit() {
      this.handleEdit.bind(this);
    },
    generateWorks() {
      this.generateWorks.bind(this);
    },
    setStreak() {
      this.setStreak.bind(this);
    },
  };

  resetMessage = () => {
    this.setState({ message: "", isShow: false });
  };

  setMessage = (str) => {
    this.setState({ message: str, isShow: true });
  };

  setStreak = (streak) => {
    console.log("streak is", streak);
    this.setState({ streak });
  };

  setStreakComplete = (boolean) => {
    this.setState({ streakComplete: boolean });
  };

  setUserId = (userid) => {
    this.setState({ userid });
    localStorage.setItem("userid", userid);
  };
  handleEdit = (e, title, content, wordcount) => {
    let id = e.target.id.slice(9, e.target.id.length);
    this.setState({
      currentWorkId: id,
      boxTitle: title,
      boxContent: content,
      wordCount: wordcount,
    });
  };

  generateWorks = () => {
    IterateApi.get(this.state.userid).then((works) => this.setState({ works }));
  };

  handleDelete(e) {
    let id = e.target.id.slice(8, e.target.id.length);
    IterateApi.delete(id).then(() => this.generateWorks());
  }

  handleKeypress = (event) => {
    let eventString = event.target.value;
    let liveWordCount = eventString.split(" ").length;
    this.setState({
      wordCount: liveWordCount,
    });
    //post and update streak via back-end
    if (
      this.state.wordCount >= this.state.DailyWritingGoal &&
      !this.state.streakComplete
    ) {
      let novelCount = 80000 / this.state.DailyWritingGoal;
      let str = `Congrats! If you wrote like this every day, you would finish a novel
      in ${novelCount} days!`;
      this.setMessage(str);
      this.setStreakComplete(true);

      streakService.postStreak(this.state.userid).then((res) => {
        this.setStreak(res.returnStreak);
      });
    }
    this.handleAnimationKeypress(event);
    this.stopAnimate();
  };

  handleDailyGoalSelector = (event) => {
    let getWordCount = event.target.value.slice(0, 4);
    let goalWordCount = parseInt(getWordCount, 10);
    this.setState({
      DailyWritingGoal: goalWordCount,
      goalSelector: event.target.value,
    });
  };

  handleContentChange = (event) => {
    this.setState({
      boxContent: event.target.value,
    });
  };

  handleLogging = () => {
    this.setState({ isLoggedIn: TokenService.hasAuthToken() });
  };

  handleTitleChange = (event) => {
    this.setState({
      boxTitle: event.target.value,
    });
  };

  handleSave = (event) => {
    event.preventDefault();
    if (this.state.currentWorkId === null) {
      const newWorks = {
        title: this.state.boxTitle,
        content: this.state.boxContent,
        wordcount: this.state.wordCount,
        user_id: this.state.userid || localStorage.getItem("userid"),
      };
      return IterateApi.post(newWorks).then((res) =>
        res.work
          ? this.setMessage("New work saved!")
          : this.setMessage(res.error)
      );
    }
    if (this.state.currentWorkdId !== null) {
      let id = this.state.currentWorkId;
      const updatedObj = {
        title: this.state.boxTitle,
        content: this.state.boxContent,
        wordcount: this.state.wordCount,
        user_id: this.state.userid || localStorage.getItem("userid"),
      };
      IterateApi.patch(id, updatedObj);
    }
  };

  handleStartNew = (event) => {
    this.handleSave(event);
    this.setState({
      boxTitle: "",
      boxContent: "",
      wordCount: 0,
      currentWorkId: null,
    });
  };

  handleAnimationKeypress = (event) => {
    this.setState({
      animationClassName: "animate",
    });
  };

  stopAnimate = () => {
    setTimeout(() => {
      this.setState({
        animationClassName: "",
      });
    }, 4000);
  };
  render() {
    const value = {
      isLoggedIn: this.state.isLoggedIn,
      userid: this.state.userid || localStorage.getItem("userid"),
      works: this.state.works,
      wordCount: this.state.wordCount,
      DailyWritingGoal: this.state.DailyWritingGoal,
      streak: this.state.streak,
      boxTitle: this.state.boxTitle,
      boxContent: this.state.boxContent,
      goalSelector: this.state.goalSelector,
      currentWorkId: this.state.currentWorkId,
      message: this.state.message,
      isShow: this.state.isShow,
      streakComplete: this.state.streakComplete,
      handleSave: this.handleSave,
      handleTitleChange: this.handleTitleChange,
      handleContentChange: this.handleContentChange,
      handleDailyGoalSelector: this.handleDailyGoalSelector,
      handleKeypress: this.handleKeypress,
      handleStartNew: this.handleStartNew,
      handleDelete: this.handleDelete,
      handleEdit: this.handleEdit,
      handleAnimationKeypress: this.handleAnimationKeypress,
      animationClassName: this.state.animationClassName,
      stopAnimate: this.stopAnimate,
      generateWorks: this.generateWorks,
      handleLogging: this.handleLogging,
      setUserId: this.setUserId,
      setStreak: this.setStreak,
      setMessage: this.setMessage,
      resetMessage: this.resetMessage,
      setStreakComplete: this.setStreakComplete,
    };
    return (
      <UserContext.Provider value={value}>
        {this.props.children}
      </UserContext.Provider>
    );
  }
}
