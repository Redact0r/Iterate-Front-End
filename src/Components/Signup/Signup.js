import React, { Component } from "react";
import AuthApiService from "../../services/auth-service";
import UserContext from "../../Context/UserContext";

class Signup extends Component {
  static contextType = UserContext;
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  state = { message: "", password: "", confirmPass: "", passwordsMatch: false };

  handleRegistrationSuccess = (user) => {
    const { history } = this.props;
    history.push("/login");
  };

  handleInitialPassword = (ev) => {
    ev.preventDefault();
    let password = ev.target.value;
    this.setState({ password });
    password === this.state.confirmPass
      ? this.setState({ message: "", passwordsMatch: true })
      : this.setState({
          message: "Passwords do not match",
          passwordsMatch: false,
        });
  };

  handlePasswordMatch = (ev) => {
    ev.preventDefault();
    let confirmPass = ev.target.value;
    this.setState({});
    confirmPass === this.state.password
      ? this.setState({ message: "", passwordsMatch: true })
      : this.setState({
          message: "Passwords do not match",
          passwordsMatch: false,
        });
  };

  handleSubmit = (ev) => {
    ev.preventDefault();
    const {
      signup_fullname,
      signup_nickname,
      signup_username,
      signup_password,
    } = ev.target;
    this.setState({ error: null });
    AuthApiService.postUser({
      user_name: signup_username.value,
      password: signup_password.value,
      full_name: signup_fullname.value,
      nickname: signup_nickname.value,
    })
      .then((user) => {
        signup_fullname.value = "";
        signup_nickname.value = "";
        signup_username.value = "";
        signup_password.value = "";
        this.handleRegistrationSuccess();
      })
      .catch((res) => {
        this.context.setMessage(res.error);
      });
  };

  render() {
    return (
      <div>
        <form onSubmit={(e) => this.handleSubmit(e)} className="signup-form">
          <label htmlFor="signup_username">Username</label>
          <input
            id="signup_username"
            type="text"
            className="user signup username"
            placeholder="Username"
          />
          <label htmlFor="signup_fullname">Full Name</label>
          <input
            id="signup_fullname"
            type="text"
            className="user login name"
            placeholder="Full Name"
          />
          <label htmlFor="signup_nickname">Nickname</label>
          <input
            id="signup_nickname"
            type="text"
            className="user signup nickname"
            placeholder="Nickname"
          />
          <label htmlFor="password">Password</label>
          <input
            id="signup_password"
            name="password"
            type="password"
            className="user signup pass"
            placeholder="Password"
            onChange={(ev) => this.handleInitialPassword(ev)}
          />
          <label htmlFor="confirm-password">Confirm Password</label>
          <input
            id="confirm_password"
            name="confirm"
            type="password"
            className="user signup confirm pass"
            placeholder="Confirm Password"
            onChange={(ev) => this.handlePasswordMatch(ev)}
          />
          {this.state.message}
          <p>
            Passwords must be at least 8 letters, contain one uppercase letter,
            special character, and number
          </p>
          <button
            className="signup-btn"
            type="submit"
            disabled={!this.state.passwordsMatch}
          >
            Sign Up
          </button>
        </form>
      </div>
    );
  }
}

export default Signup;
