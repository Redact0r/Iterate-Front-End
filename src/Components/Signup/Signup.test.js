import React from "react";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Signup from "./Signup";
import UserContext from "../../Context/UserContext";

configure({ adapter: new Adapter() });

describe(`Adds Signup to dom without crashing`, () => {
  it("renders without crashing", () => {
    const wrapper = shallow(
      <UserContext.Provider>
        <Signup />
      </UserContext.Provider>
    ).find(".signup-form");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
