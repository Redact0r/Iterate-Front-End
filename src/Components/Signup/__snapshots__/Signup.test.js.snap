// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds Signup to dom without crashing renders without crashing 1`] = `null`;
